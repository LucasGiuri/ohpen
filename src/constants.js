export const colors = {
  white: '#FFF',
  black: '#000',
  bgGrey: '#eee',
  lightGrey: '#bebebe',
  green: '#4fbe79'
};

export const radius = {
  small: '4px',
  large: '15px',
  rounded: '50%'
};

export const mediaSizes = {
  mobile: 'max-width: 800px',
  ipad: 'max-width: 1200px'
};

export const mockData = [
  { id: 1, title: 'subscription1', checked: true, content: '1234567' },
  { id: 2, title: 'subscription2', checked: false, content: 'Lorem' },
  { id: 3, title: 'subscription3', checked: true, content: 'ipsum' },
  { id: 4, title: 'subscription4', checked: false, content: '1234567 1234567 1234567 1234567 1234567 1234567 1234567 1234567 1234567' },
  { id: 5, title: 'subscription5', checked: true, content: 'qwertyu asdfdgg zcxvbvn aesdg qewrte zcxxvc' },
  { id: 6, title: 'subscription6', checked: false, content: '1234567 test123 test123 test123 test123  test123 test123 test123 test123' },
  { id: 7, title: 'subscription7', checked: true, content: '1234567 123 123 123 123 123' }
];

export const typesOfUsers = ['HR', 'Administrator'];

export const fakeData = [
  { id: 1, username: 'Lucas', subscriptions: mockData, type: 'Administrator' },
  { id: 2, username: 'Marco', subscriptions: mockData, type: 'HR' }
];

