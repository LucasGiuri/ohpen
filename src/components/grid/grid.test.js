import Grid from './grid';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { mockData } from '../../constants';

describe('Grid', () => {
  const dataTestPrefix = 'dataTestPrefix';

  describe('Snapshots', () => {
    it('default snapshot render', () => {
      const tree = renderer.create(<Grid data={mockData} />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('with custom dataPrefix snapshot render', () => {
      const tree = renderer.create(<Grid data={mockData} dataTestPrefix={dataTestPrefix} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
