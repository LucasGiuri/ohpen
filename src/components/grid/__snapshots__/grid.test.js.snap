// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid Snapshots default snapshot render 1`] = `
<div
  className="sc-iBPTik eXxKsk"
  data-test="grid-grid"
>
  <div
    className="sc-fubCzh kPWXjr"
  >
    <div
      className="sc-bdfBQB fINfZT"
      data-test="card-card"
    >
      <div
        className="sc-gsTEea bMmgDr"
      >
        <span
          className="sc-dlfnuX cvZFiX"
          data-test="card-card-title"
        >
          SUBSCRIPTION1
        </span>
        <div
          className="sc-eCstlR jXyqPy"
        >
          <input
            checked={true}
            className="sc-gKseQn gLUVYz"
            type="checkbox"
          />
          <label
            className="sc-jSgvzq iZDWJJ"
            htmlFor="checkbox"
            onClick={[Function]}
          />
        </div>
      </div>
      <div
        className="sc-hKgJUU dbFtMW"
        data-test="card-card-content"
      >
        1234567
      </div>
    </div>
  </div>
  <div
    className="sc-fubCzh kPWXjr"
  >
    <div
      className="sc-bdfBQB fINfZT"
      data-test="card-card"
    >
      <div
        className="sc-gsTEea bMmgDr"
      >
        <span
          className="sc-dlfnuX cvZFiX"
          data-test="card-card-title"
        >
          SUBSCRIPTION2
        </span>
        <div
          className="sc-eCstlR jXyqPy"
        >
          <input
            checked={false}
            className="sc-gKseQn gLUVYz"
            type="checkbox"
          />
          <label
            className="sc-jSgvzq iZDWJJ"
            htmlFor="checkbox"
            onClick={[Function]}
          />
        </div>
      </div>
      <div
        className="sc-hKgJUU dbFtMW"
        data-test="card-card-content"
      >
        Lorem
      </div>
    </div>
  </div>
  <div
    className="sc-fubCzh kPWXjr"
  >
    <div
      className="sc-bdfBQB fINfZT"
      data-test="card-card"
    >
      <div
        className="sc-gsTEea bMmgDr"
      >
        <span
          className="sc-dlfnuX cvZFiX"
          data-test="card-card-title"
        >
          SUBSCRIPTION3
        </span>
        <div
          className="sc-eCstlR jXyqPy"
        >
          <input
            checked={true}
            className="sc-gKseQn gLUVYz"
            type="checkbox"
          />
          <label
            className="sc-jSgvzq iZDWJJ"
            htmlFor="checkbox"
            onClick={[Function]}
          />
        </div>
      </div>
      <div
        className="sc-hKgJUU dbFtMW"
        data-test="card-card-content"
      >
        ipsum
      </div>
    </div>
  </div>
  <div
    className="sc-fubCzh kPWXjr"
  >
    <div
      className="sc-bdfBQB fINfZT"
      data-test="card-card"
    >
      <div
        className="sc-gsTEea bMmgDr"
      >
        <span
          className="sc-dlfnuX cvZFiX"
          data-test="card-card-title"
        >
          SUBSCRIPTION4
        </span>
        <div
          className="sc-eCstlR jXyqPy"
        >
          <input
            checked={false}
            className="sc-gKseQn gLUVYz"
            type="checkbox"
          />
          <label
            className="sc-jSgvzq iZDWJJ"
            htmlFor="checkbox"
            onClick={[Function]}
          />
        </div>
      </div>
      <div
        className="sc-hKgJUU dbFtMW"
        data-test="card-card-content"
      >
        1234567 1234567 1234567 1234567 1234567 1234567 1234567 1234567 1234567
      </div>
    </div>
  </div>
  <div
    className="sc-fubCzh kPWXjr"
  >
    <div
      className="sc-bdfBQB fINfZT"
      data-test="card-card"
    >
      <div
        className="sc-gsTEea bMmgDr"
      >
        <span
          className="sc-dlfnuX cvZFiX"
          data-test="card-card-title"
        >
          SUBSCRIPTION5
        </span>
        <div
          className="sc-eCstlR jXyqPy"
        >
          <input
            checked={true}
            className="sc-gKseQn gLUVYz"
            type="checkbox"
          />
          <label
            className="sc-jSgvzq iZDWJJ"
            htmlFor="checkbox"
            onClick={[Function]}
          />
        </div>
      </div>
      <div
        className="sc-hKgJUU dbFtMW"
        data-test="card-card-content"
      >
        qwertyu asdfdgg zcxvbvn aesdg qewrte zcxxvc
      </div>
    </div>
  </div>
  <div
    className="sc-fubCzh kPWXjr"
  >
    <div
      className="sc-bdfBQB fINfZT"
      data-test="card-card"
    >
      <div
        className="sc-gsTEea bMmgDr"
      >
        <span
          className="sc-dlfnuX cvZFiX"
          data-test="card-card-title"
        >
          SUBSCRIPTION6
        </span>
        <div
          className="sc-eCstlR jXyqPy"
        >
          <input
            checked={false}
            className="sc-gKseQn gLUVYz"
            type="checkbox"
          />
          <label
            className="sc-jSgvzq iZDWJJ"
            htmlFor="checkbox"
            onClick={[Function]}
          />
        </div>
      </div>
      <div
        className="sc-hKgJUU dbFtMW"
        data-test="card-card-content"
      >
        1234567 test123 test123 test123 test123  test123 test123 test123 test123
      </div>
    </div>
  </div>
  <div
    className="sc-fubCzh kPWXjr"
  >
    <div
      className="sc-bdfBQB fINfZT"
      data-test="card-card"
    >
      <div
        className="sc-gsTEea bMmgDr"
      >
        <span
          className="sc-dlfnuX cvZFiX"
          data-test="card-card-title"
        >
          SUBSCRIPTION7
        </span>
        <div
          className="sc-eCstlR jXyqPy"
        >
          <input
            checked={true}
            className="sc-gKseQn gLUVYz"
            type="checkbox"
          />
          <label
            className="sc-jSgvzq iZDWJJ"
            htmlFor="checkbox"
            onClick={[Function]}
          />
        </div>
      </div>
      <div
        className="sc-hKgJUU dbFtMW"
        data-test="card-card-content"
      >
        1234567 123 123 123 123 123
      </div>
    </div>
  </div>
</div>
`;

exports[`Grid Snapshots with custom dataPrefix snapshot render 1`] = `
<div
  className="sc-iBPTik eXxKsk"
  data-test="dataTestPrefix-grid"
>
  <div
    className="sc-fubCzh kPWXjr"
  >
    <div
      className="sc-bdfBQB fINfZT"
      data-test="card-card"
    >
      <div
        className="sc-gsTEea bMmgDr"
      >
        <span
          className="sc-dlfnuX cvZFiX"
          data-test="card-card-title"
        >
          SUBSCRIPTION1
        </span>
        <div
          className="sc-eCstlR jXyqPy"
        >
          <input
            checked={true}
            className="sc-gKseQn gLUVYz"
            type="checkbox"
          />
          <label
            className="sc-jSgvzq iZDWJJ"
            htmlFor="checkbox"
            onClick={[Function]}
          />
        </div>
      </div>
      <div
        className="sc-hKgJUU dbFtMW"
        data-test="card-card-content"
      >
        1234567
      </div>
    </div>
  </div>
  <div
    className="sc-fubCzh kPWXjr"
  >
    <div
      className="sc-bdfBQB fINfZT"
      data-test="card-card"
    >
      <div
        className="sc-gsTEea bMmgDr"
      >
        <span
          className="sc-dlfnuX cvZFiX"
          data-test="card-card-title"
        >
          SUBSCRIPTION2
        </span>
        <div
          className="sc-eCstlR jXyqPy"
        >
          <input
            checked={false}
            className="sc-gKseQn gLUVYz"
            type="checkbox"
          />
          <label
            className="sc-jSgvzq iZDWJJ"
            htmlFor="checkbox"
            onClick={[Function]}
          />
        </div>
      </div>
      <div
        className="sc-hKgJUU dbFtMW"
        data-test="card-card-content"
      >
        Lorem
      </div>
    </div>
  </div>
  <div
    className="sc-fubCzh kPWXjr"
  >
    <div
      className="sc-bdfBQB fINfZT"
      data-test="card-card"
    >
      <div
        className="sc-gsTEea bMmgDr"
      >
        <span
          className="sc-dlfnuX cvZFiX"
          data-test="card-card-title"
        >
          SUBSCRIPTION3
        </span>
        <div
          className="sc-eCstlR jXyqPy"
        >
          <input
            checked={true}
            className="sc-gKseQn gLUVYz"
            type="checkbox"
          />
          <label
            className="sc-jSgvzq iZDWJJ"
            htmlFor="checkbox"
            onClick={[Function]}
          />
        </div>
      </div>
      <div
        className="sc-hKgJUU dbFtMW"
        data-test="card-card-content"
      >
        ipsum
      </div>
    </div>
  </div>
  <div
    className="sc-fubCzh kPWXjr"
  >
    <div
      className="sc-bdfBQB fINfZT"
      data-test="card-card"
    >
      <div
        className="sc-gsTEea bMmgDr"
      >
        <span
          className="sc-dlfnuX cvZFiX"
          data-test="card-card-title"
        >
          SUBSCRIPTION4
        </span>
        <div
          className="sc-eCstlR jXyqPy"
        >
          <input
            checked={false}
            className="sc-gKseQn gLUVYz"
            type="checkbox"
          />
          <label
            className="sc-jSgvzq iZDWJJ"
            htmlFor="checkbox"
            onClick={[Function]}
          />
        </div>
      </div>
      <div
        className="sc-hKgJUU dbFtMW"
        data-test="card-card-content"
      >
        1234567 1234567 1234567 1234567 1234567 1234567 1234567 1234567 1234567
      </div>
    </div>
  </div>
  <div
    className="sc-fubCzh kPWXjr"
  >
    <div
      className="sc-bdfBQB fINfZT"
      data-test="card-card"
    >
      <div
        className="sc-gsTEea bMmgDr"
      >
        <span
          className="sc-dlfnuX cvZFiX"
          data-test="card-card-title"
        >
          SUBSCRIPTION5
        </span>
        <div
          className="sc-eCstlR jXyqPy"
        >
          <input
            checked={true}
            className="sc-gKseQn gLUVYz"
            type="checkbox"
          />
          <label
            className="sc-jSgvzq iZDWJJ"
            htmlFor="checkbox"
            onClick={[Function]}
          />
        </div>
      </div>
      <div
        className="sc-hKgJUU dbFtMW"
        data-test="card-card-content"
      >
        qwertyu asdfdgg zcxvbvn aesdg qewrte zcxxvc
      </div>
    </div>
  </div>
  <div
    className="sc-fubCzh kPWXjr"
  >
    <div
      className="sc-bdfBQB fINfZT"
      data-test="card-card"
    >
      <div
        className="sc-gsTEea bMmgDr"
      >
        <span
          className="sc-dlfnuX cvZFiX"
          data-test="card-card-title"
        >
          SUBSCRIPTION6
        </span>
        <div
          className="sc-eCstlR jXyqPy"
        >
          <input
            checked={false}
            className="sc-gKseQn gLUVYz"
            type="checkbox"
          />
          <label
            className="sc-jSgvzq iZDWJJ"
            htmlFor="checkbox"
            onClick={[Function]}
          />
        </div>
      </div>
      <div
        className="sc-hKgJUU dbFtMW"
        data-test="card-card-content"
      >
        1234567 test123 test123 test123 test123  test123 test123 test123 test123
      </div>
    </div>
  </div>
  <div
    className="sc-fubCzh kPWXjr"
  >
    <div
      className="sc-bdfBQB fINfZT"
      data-test="card-card"
    >
      <div
        className="sc-gsTEea bMmgDr"
      >
        <span
          className="sc-dlfnuX cvZFiX"
          data-test="card-card-title"
        >
          SUBSCRIPTION7
        </span>
        <div
          className="sc-eCstlR jXyqPy"
        >
          <input
            checked={true}
            className="sc-gKseQn gLUVYz"
            type="checkbox"
          />
          <label
            className="sc-jSgvzq iZDWJJ"
            htmlFor="checkbox"
            onClick={[Function]}
          />
        </div>
      </div>
      <div
        className="sc-hKgJUU dbFtMW"
        data-test="card-card-content"
      >
        1234567 123 123 123 123 123
      </div>
    </div>
  </div>
</div>
`;
